planning/behavior_path_planner/src/scene_module/avoidance/avoidance_module.cpp

```diff
@@ -3350,23 +3351,64 @@ void AvoidanceModule::insertPrepareVelocity(ShiftedPath & shifted_path) const
 {
   const auto & data = avoidance_data_;
 
+  // do nothing if there is no avoidance target.
   if (data.target_objects.empty()) {
     return;
   }
 
-  if (!!data.stop_target_object) {
-    if (data.stop_target_object.get().reason != AvoidanceDebugFactor::TOO_LARGE_JERK) {
-      return;
-    }
+  // insert slow down speed only when the avoidance maneuver is not initiated.
+  if (data.avoiding_now) {
+    return;
   }
 
-  if (data.avoiding_now) {
+  // insert slow down speed only when no shift line is approved.
+  if (!path_shifter_.getShiftLines().empty()) {
     return;
   }
 
-  const auto decel_distance = helper_.getFeasibleDecelDistance(0.0);
-  utils::avoidance::insertDecelPoint(
-    getEgoPosition(), decel_distance, 0.0, shifted_path.path, slow_pose_);
+  const auto object = data.target_objects.front();
+
+  // calculate shift length for front object.
+  const auto & vehicle_width = planner_data_->parameters.vehicle_width;
+  const auto t = utils::getHighestProbLabel(object.object.classification);
```

Comment:
```suggestion
  const auto object_type = utils::getHighestProbLabel(object.object.classification);
```

Reply:
Rename all same variable in https://github.com/autowarefoundation/autoware.universe/pull/4141/commits/7dfd7569d6325a639d635e58e50bd63fd6a8c0cb.